modules/config/URLProcessor.js

```diff
@@ -1,69 +1,94 @@
+
+/**
+ *  URL params with this prefix should be merged to logging config.
+ */
+const LOGGING_CONFIG_PREFIX = 'loggingConfig.';
+
+/**
+ * Config keys to be ignored.
+ *
+ * @type Set
+ */
+const KEYS_TO_IGNORE = new Set([
+    'analyticsScriptUrls',
+    'callStatsCustomScriptUrl'
+]);
+
+/**
+ * Convert 'URL_PARAMS' to JSON object
+ * We have:
+ * {
+ *      "config.disableAudioLevels": false,
+ *      "config.channelLastN": -1,
+ *      "interfaceConfig.APP_NAME": "Jitsi Meet"
+ * }
+ * We want to have:
+ * {
+ *      "config": {
+ *          "disableAudioLevels": false,
+ *          "channelLastN": -1
+ *      },
+ *      interfaceConfig: {
+ *          "APP_NAME": "Jitsi Meet"
+ *      }
+ * }
+ */
+export default {
+    setConfigParametersFromUrl() {
+        const configJSON = {
             config: {},
             interfaceConfig: {},
             loggingConfig: {}
         };
-        for (var key in params) {
-            if (typeof key !== "string") {
-                logger.warn("Invalid config key: ", key);
-                continue;
-            }
-            var confObj = null, confKey;
-            if (key.indexOf("config.") === 0) {
-                confObj = configJSON.config;
-                confKey = key.substr("config.".length);
 
-                // prevent passing some parameters which can inject scripts
-                if (confKey === 'analyticsScriptUrls'
-                    || confKey === 'callStatsCustomScriptUrl')
-                    continue;
+        for (const key in URL_PARAMS) {
+            if (typeof key === 'string') {
+                let confObj = null;
+                let confKey;
 
-            } else if (key.indexOf("interfaceConfig.") === 0) {
-                confObj = configJSON.interfaceConfig;
-                confKey = key.substr("interfaceConfig.".length);
-            } else if (key.indexOf("loggingConfig.") === 0) {
-                confObj = configJSON.loggingConfig;
-                confKey = key.substr("loggingConfig.".length);
-            }
+                if (key.indexOf(CONFIG_PREFIX) === 0) {
+                    confObj = configJSON.config;
+                    confKey = key.substr(CONFIG_PREFIX.length);
 
-            if (!confObj)
-                continue;
+                    // prevent passing some parameters which can inject scripts
+                    if (confObj && !KEYS_TO_IGNORE.has(confKey)) {
+                        confObj[confKey] = URL_PARAMS[key];
+                    }
 
-            confObj[confKey] = params[key];
+                } else if (key.indexOf(INTERFACE_CONFIG_PREFIX) === 0) {
+                    confObj = configJSON.interfaceConfig;
+                    confKey
+                        = key.substr(INTERFACE_CONFIG_PREFIX.length);
+                } else if (key.indexOf(LOGGING_CONFIG_PREFIX) === 0) {
+                    confObj = configJSON.loggingConfig;
+                    confKey = key.substr(LOGGING_CONFIG_PREFIX.length);
+                }
```

Comment:
The branches for `interfaceConfig` and `loggingConfig` have become incomplete because nothing after the assignments to `confObj` and `confKey` to make a difference in `configJSON`.